clvm/serialize.py

```diff
@@ -153,9 +156,14 @@ def atom_to_byte_iterator(as_atom: bytes) -> typing.Iterator[bytes]:
 
 
 def sexp_to_stream(
-    sexp: CLVMStorage, f: typing.BinaryIO, *, allow_backrefs: bool = False
+    sexp: CLVMStorage, f: typing.BinaryIO, *, allow_backrefs: bool = False, max_size: Optional[int] = None
 ) -> None:
+    remaining = max_size
```

Comment:
if you make `max_size` non-optional, you can just count down that variable instead of making a copy